net: openthread: bring back `LOG_MODE_MINIMAL` support

edmont opened the pull request:
A previous commit fixed OpenThread logging when `LOG=n`, but
introduced regression when `LOG_MODE_MINIMAL=y`. This commit
fixes the latest.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

edmont commented:
Referred PR: https://github.com/zephyrproject-rtos/zephyr/pull/41957

rlubos commented on subsys/net/lib/openthread/platform/logging.c:
```diff
@@ -39,6 +39,28 @@ static inline int log_translate(otLogLevel aLogLevel)
 }
 
+#if defined(CONFIG_LOG_MODE_MINIMAL)
+uint32_t log_count_args(const char *fmt)
```
Maybe instead of relying on the logger configuration, we could have an internal, `static` utility function to count arguments, used in either case?

edmont replied:
Sure, sounds more future proof.

jukkar commented on subsys/net/lib/openthread/platform/logging.c:
```diff
@@ -39,13 +39,33 @@ static inline int log_translate(otLogLevel aLogLevel)
 #if defined(CONFIG_LOG)
 	int level = log_translate(aLogLevel);
-	uint32_t args_num = log_count_args(aFormat);
+	uint32_t args_num = count_args(aFormat);
```
Is there compiler warning about unused function if CONFIG_LOG is not set?

edmont replied:
It does, thanks for catching. Fixed now.

edmont committed:
net: openthread: bring back `LOG_MODE_MINIMAL` support

A previous commit fixed OpenThread logging when `LOG=n`, but
introduced regression when `LOG_MODE_MINIMAL=y`. This commit
fixes the latest.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

rlubos reviewed (approved):
Thanks

edmont commented:
Closing as it's included in https://github.com/zephyrproject-rtos/zephyr/pull/42307
